quic/s2n-quic-transport/src/ack/pending_ack_ranges.rs

```diff
@@ -14,18 +14,11 @@ pub struct PendingAckRanges {
     ack_ranges: AckRanges,
     ecn_counts: EcnCounts,
     ack_delay: Duration,
+    /// The path for which to aggregate ACKs
+    pub current_active_path: Option<path::Id>,
```

Comment:
Would it be cleaner to have `pending_ack_ranges: Option<PendingAckRanges>` on the application space, so you have `path::Id` be set when the first ack comes in? Seems like you have a ton of debug assertions to ensure the state of PendingAckRanges is correct, might be better if you can avoid making it possible to get into a bad state

Reply:
We dont want to reallocate `PendingAckRanges{AckRanges{IntervalSet}}` each time and therefore the current design. Let me think if there is a clear way to capture this current path per round.

Reply:
We should continue with the current solution since we want to avoid reallocating all of PendingAckRanges each round.

Memory cost:
- Option<Pending>: would reallocate the VecDeque + Ecn + Duration + Path (88 bytes)
- Option<currentPath>: would only reallocate the u8 (8 bytes)

Instruction cost: (based on [this](https://stackoverflow.com/a/36512638) answer)
- Option<Pending>: since VecDeque is heap allocated (100s-1000s of instructions)
- Option<currentPath>: since u8 is stack allocated (4 instructions)